chart/templates/statsd/statsd-deployment.yaml

```diff
@@ -89,7 +88,7 @@ spec:
         - name: statsd
           image: {{ template "statsd_image" . }}
           imagePullPolicy: {{ .Values.images.statsd.pullPolicy }}
-          securityContext: {{ $containerSecurityContext | nindent 12 }}
+          securityContext: {{- toYaml .Values.statsd.securityContexts.container | nindent 12 }}
```

Comment:
This breaks backcompat on `stastd.securityContext`.

Reply:
@jedcunningham actually this line is not present at all in the latest helm chart release, it was added in #31043, see https://github.com/apache/airflow/pull/31043/files#diff-9afea6c64dc71d0fab91a1761ff0b2a9334a6751f5a2026c6d0ba7f57ac47ea1R92 - so I don't think there'll be any issues with backwards compatibility. Or do you think I should add something like the logic `<node>.securityContexts.container  -> allowPrivilegesEscalation: false, capabilities.drop: [ALL]` anyway? 
Actually thinking about it adding the logic of `<node>.securityContexts.container  -> allowPrivilegesEscalation: false, capabilities.drop: [ALL]` looks like a good idea, just to be in line with the other containers of the chart that also add `allowPrivilegesEscalation: false, capabilities.drop: [ALL]` by default to be compliant with restricted pod security standards.

Reply:
Added the logic that by default `allowPrivilegesEscalation: false, capabilities.drop: [ALL]` will be set for redis/statsd.

Reply:
Also, if by backcompat you mean that `stastd.securityContext` shoud end up in statsd container security context - I disagree, in the current dev version `stastd.securityContext` is not passed to statsd container security context, see https://github.com/apache/airflow/blob/main/chart/templates/_helpers.yaml#L842 and https://github.com/apache/airflow/blob/main/chart/templates/statsd/statsd-deployment.yaml#L30 . And as I said before, in previous releases statsd container security context wasn't even present.